frontend/wallet/src/render/views/Transactions/Transactions.re

```diff
@@ -163,49 +125,58 @@ let make = () => {
-                      <TransactionCell transaction />
-                    </div>,
-                extractTransactions(data),
-              )
-              |> React.array}
-             {!isFetchingMore
-                ? <Waypoint
-                    onEnter={_ => {
-                      setFetchingMore(_ => true);
-                      let moreTransactions =
-                        Transactions.make(~publicKey=activeWalletKey, ());
-                      let _ =
-                        response.fetchMore(
-                          ~variables=moreTransactions##variables,
-                          ~updateQuery,
-                          (),
-                        )
-                        |> Js.Promise.then_(_ => {
-                             setFetchingMore(_ => false);
-                             Js.Promise.resolve();
-                           });
-                      ();
-                    }}
-                    topOffset="100px"
-                  />
-                : <div
-                    className=Css.(style([margin2(~v=`rem(1.5), ~h=`auto)]))>
-                    <Loader />
-                  </div>}
-           </div>
-         }}
-    </TransactionsQuery>
+    {switch (activeWallet) {
+     | Some(pubkey) =>
+       let transactionQuery =
+         TransactionsQueryString.make(
+           ~publicKey=PublicKey.toString(pubkey),
+           ~after="",
+           (),
+         );
+       <TransactionsQuery variables=transactionQuery##variables>
+         (
+           response =>
+             switch (response.result) {
+             | Loading => <Loader.Page> <Loader /> </Loader.Page>
+             | Error(err) => React.string(err##message) /* TODO format this error message */
+             | Data(data) =>
+               let transactions = extractTransactions(data);
+               switch (Array.length(transactions)) {
+               | 0 =>
+                 <div className=Styles.alertContainer>
+                   <Alert
+                     kind=`Info
+                     message="You don't have any coda in this wallet."
+                   />
+                 </div>
+               | _ =>
+                 <TransactionsList
+                   transactions
+                   onLoadMore={() => {
+                     let moreTransactions =
+                       TransactionsQueryString.make(
+                         ~publicKey=PublicKey.toString(pubkey),
+                         (),
+                       );
+
+                     response.fetchMore(
+                       ~variables=moreTransactions##variables,
+                       ~updateQuery,
+                       (),
+                     );
+                   }}
+                 />
+               };
+             }
+         )
+       </TransactionsQuery>;
+     | None =>
+       <div className=Styles.alertContainer>
+         <Alert
+           message="Select a wallet from the side bar to view related transactions."
```

Comment:
Don't have copy for this, so it's just placeholder. We could alternatively forward you to the first active wallet? Dunno what we wanna do here.

Reply:
Probably want to forward to first wallet if we can, longterm